enclave/core/sgx/enter.S

```diff
@@ -39,23 +40,26 @@ oe_enter:
 .cfi_startproc
 
 .save_host_registers:
-    // Backup the current host rbp and rsp to previous.
+    // Backup the current host rbp, rsp and context to previous.
     mov %fs:td_host_rbp, %r8
     mov %r8, %fs:td_host_previous_rbp
     mov %fs:td_host_rsp, %r8
     mov %r8, %fs:td_host_previous_rsp
+    mov %fs:td_host_ecall_context, %r8
```

Comment:
It looks good the way it is, but I was wondering whether **td_host_ecall_context** could just be a **__thread** variable (TLS). Then I suppose this would become:

```
mov    %fs:td_host_previous_rsp@tpoff, %r8
```

And if so, should we transition to using thread local storage for other fields?

Reply:
Yes. TLS in not initialized at this point. I have created an issue to remove use of td_t and use thread-local variables instead. #2554 
Based on initial experiments, it should be quite possible to switch to tls variables.